### PR TITLE
fix: 🐛 pass token when retrieving the split names

### PR DESCRIPTION
**Pull request by severo**

### CODE REVIEW

File: src/datasets/inspect.py [hunk 1]
```diff
@@ -271,9 +271,12 @@ def get_dataset_split_names(
     )
     if builder.info.splits is None:
         try:
+            download_config.use_auth_token = use_auth_token
```
Comment (severo) [inline src/datasets/inspect.py:274]: by the way, I'm not sure if I should do this, or if `get_dataset_split_names` should be called with both `use_auth_token` and `download_config` parameters (seems overly complicated)

**Comment by severo** [comment]
Currently, it does not work with https://huggingface.co/datasets/mozilla-foundation/common_voice_7_0/blob/main/common_voice_7_0.py#L146 (which was the goal), because `dl_manager.download_config.use_auth_token` is ignored, and the token is required to be passed using `huggingface-cli login`

File: src/datasets/inspect.py (hunk 1 above)
Comment (lhoestq) [reply src/datasets/inspect.py:274]: Simply passing `download_config` is ok :)

Comment (lhoestq) [inline src/datasets/inspect.py:274]:
```suggestion
            download_config = download_config.copy() if download_config else DownloadConfig()
            download_config.use_auth_token = use_auth_token
```

Comment (lhoestq) [reply src/datasets/inspect.py:274]: in case `download_config` is `None`

**Review by lhoestq — COMMENTED** [review]
Thank you ! It looks all good :)
Just one comment in case `download_config` is `None`:

**Comment by severo** [comment]
> Simply passing download_config is ok :)
hmm, I prefer only passing use_auth_token. But the question is more: is it correct, in the (convoluted) case if `download_config.use_auth_token` exists and is different from `use_auth_token`? Which one should be used?

**Comment by lhoestq** [comment]
If both are passed, `use_auth_token` should have the priority (more specific parameters have the higher priority)

**Review by lhoestq — APPROVED** [review]
Thanks ! It looks all good :) Feel free to merge if it's good for you

**Pull request merged**